client/orm/db.go

```diff
@@ -1372,39 +1379,42 @@ func (d *dbBase) readSQL(tables *dbTables, tCols []string, cond *Condition, qs *
 		_, _ = buf.WriteString(" FOR UPDATE")
 	}
 
-	query := buf.String()
-
-	d.ins.ReplaceMarks(&query)
-
-	return query, args
+	return args
 }
 
 // Count excute count sql and return count result int64.
 func (d *dbBase) Count(ctx context.Context, q dbQuerier, qs *querySet, mi *models.ModelInfo, cond *Condition, tz *time.Location) (cnt int64, err error) {
+
+	query, args := d.countSQL(qs, mi, cond, tz)
+
+	row := q.QueryRowContext(ctx, query, args...)
+	err = row.Scan(&cnt)
+	return
+}
+
+func (d *dbBase) countSQL(qs *querySet, mi *models.ModelInfo, cond *Condition, tz *time.Location) (string, []interface{}) {
 	tables := newDbTables(mi, d.ins)
 	tables.parseRelated(qs.related, qs.relDepth)
 
-	where, args := tables.getCondSQL(cond, false, tz)
-	groupBy := tables.getGroupSQL(qs.groups)
-	tables.getOrderSQL(qs.orders)
-	join := tables.getJoinSQL()
-	specifyIndexes := tables.getIndexSql(mi.Table, qs.useIndex, qs.indexes)
+	buf := buffers.Get()
+	defer buffers.Put(buf)
 
-	Q := d.ins.TableQuote()
+	if len(qs.groups) > 0 {
+		_, _ = buf.WriteString("SELECT COUNT(*) FROM (")
```

Comment:
why do you need to write string "SELECT COUNT(*) FROM " here?
I found that function `readSQL`:
```go
	} else {
		_, _ = buf.WriteString(qs.aggregate)
	}
```

Reply:
Because if grouped, each group will generate one piece of data, and multiple pieces of data will be generated. A query should be nested outside to query the number of groups.